datafusion/core/tests/dataframe/mod.rs

```diff
@@ -3325,6 +3326,74 @@ async fn unnest_columns() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn unnest_dict_encoded_columns() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    let str1 = lit(ScalarValue::Dictionary(
+        Box::new(DataType::Int32),
+        Box::new(ScalarValue::new_utf8("x")),
+    ));
+    let str2 = lit(ScalarValue::Dictionary(
+        Box::new(DataType::Int32),
+        Box::new(ScalarValue::new_utf8("y")),
+    ));
+
+    let make_array_udf_expr1 = Expr::ScalarFunction(ScalarFunction::new_udf(
+        make_array_udf(),
+        vec![col("column1")],
+    ));
```

Comment:
If we rewrote this test to use the DataFrame API I think it would belong here nicely

For example:
```suggestion
    let make_array_udf_expr1 =make_array_udf()
      .call(vec![col("column1")]));
```

Then instead of LogicalPlanBuilder, use the methods on DataFrame 🤔

Reply:
thanks, i applied the suggestion